R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,185 @@
+        # Fuel and mode will be mapped to NBS fuel and sector
+        left_join_error_no_match(trnUCD_NBS_mapping, by = c("fuel", "mode")) ->
+        L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh
+
+      # To delate the confilct category caused by Hong Kong and macau
+      L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>%
+        mutate(size.class = replace(size.class, mode == "Bus" & size.class == "All","Light Bus")) %>%
+        mutate(size.class = replace(size.class, mode == "Truck" & size.class == "Truck (0-2t)","Truck (0-6t)")) %>%
+        mutate(size.class = replace(size.class, mode == "Truck" & size.class == "Truck (2-5t)","Truck (0-6t)")) %>%
+        mutate(size.class = replace(size.class, mode == "Truck" & size.class == "Truck (5-9t)","Truck (6-14t)")) %>%
+        mutate(size.class = replace(size.class, mode == "Truck" & size.class == "Truck (9-16t)","Truck (6-14t)")) %>%
+        group_by(GCAM_region_ID, UCD_sector, mode, size.class, UCD_fuel, UCD_technology, fuel, year, EBProcess, EBMaterial) %>%
+        summarise(value = sum(value)) %>%
+        ungroup ->
+        L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh
+
+
+      # Next, extract the relevant NBS sector & fuel combinations from the full province database
+      L101.NBS_use_all_Mtce %>%
+        # Ensure within historical period
+        filter(year %in% HISTORICAL_YEARS) %>%
+        # Create concatenated list in base dataframe to match the syntax of our list above
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>%
+        # Filtering for just NBS-fuel/sector pairs
+        filter(fuel_sector %in% paste(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBProcess, L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBMaterial)) %>%
+        select(province, EBProcess, EBMaterial, sector, fuel, year, value) %>%
+        group_by(province, year) %>%
+        # First zero out NAs in years where some values are NA but not all
+        mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+        ungroup %>%
+        # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+        group_by(province, EBProcess, EBMaterial) %>%
+        mutate(value = approx_fun(year, value, rule = 2)) %>%
+        ungroup() ->
+        L154.NBS_trn_Mtce_province
+
+      # Now the province shares can be calculated
+      L154.NBS_trn_Mtce_province %>%
+        group_by(EBProcess, EBMaterial, year) %>%
+        mutate(value_share = value / sum(value, na.rm = T)) %>%
+        complete(nesting(year, sector, fuel, EBProcess, EBMaterial), province = gcamchina.PROVINCES, fill = list(value_share = 0)) %>%
+        # NAs were introduced where national values were 0. Replace NAs with zeros.
+        replace_na(list(value_share = 0)) %>%
+        select(province, EBProcess, EBMaterial, year, value_share) ->
+        L154.NBS_trn_share_province
```

Comment:
I prefer `left_join` the national values here, instead of `full_join` below.
`%>% left_join(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>% filter(year %in% HISTORICAL_YEARS) %>% mutate(fuel_sector = paste(EBProcess, EBMaterial))`